cypress/e2e/WebInterface/Test Cases/TestCaseImport/TestCaseImportviaFile.cy.ts

```diff
@@ -0,0 +1,338 @@
+let testCaseJsonTstLargeBatch03 = TestCaseJson.largeBatchTC4ImportId3
+let testCaseJsonTstLargeBatch04 = TestCaseJson.largeBatchTC4ImportId4
+let testCaseJsonTstLargeBatch05 = TestCaseJson.largeBatchTC4ImportId5
+describe('Import Test cases onto an existing measure via file', () => {
+
+    beforeEach('Login and Create Measure', () => {
+
+        CqlLibraryName = 'TestLibrary2' + Date.now()
+
+        //Create New Measure
+        CreateMeasurePage.CreateQICoreMeasureAPI(measureName, CqlLibraryName, mesureCQLPFTests)
+        OktaLogin.Login()
+        MeasuresPage.clickEditforCreatedMeasure()
+        cy.get(EditMeasurePage.cqlEditorTab).click()
+        cy.get(EditMeasurePage.cqlEditorTextBox).scrollIntoView()
+        cy.get(EditMeasurePage.cqlEditorTextBox).click().type('{enter}')
+        cy.get(EditMeasurePage.cqlEditorSaveButton).should('exist')
+        cy.get(EditMeasurePage.cqlEditorSaveButton).should('be.visible')
+        cy.get(EditMeasurePage.cqlEditorSaveButton).should('be.enabled')
+        cy.get(EditMeasurePage.cqlEditorSaveButton).click()
+        //wait for alert / succesful save message to appear
+        Utilities.waitForElementVisible(CQLEditorPage.successfulCQLSaveNoErrors, 27700)
+        cy.get(CQLEditorPage.successfulCQLSaveNoErrors).should('be.visible')
+
+
+    })
+
+    afterEach('Logout and Clean up', () => {
+
+        OktaLogin.Logout()
+        Utilities.deleteMeasure(measureName, CqlLibraryName)
+
+    })
+    it('Import new test case onto an existing measure -- single test case and verify JSON', () => {
+        //navigate to the main measures page
+        cy.get(Header.measures).click()
+
+        //click on created measure
+        MeasuresPage.clickEditforCreatedMeasure()
+
+        //click on the test case tab
+        cy.get(EditMeasurePage.testCasesTab).click()
+
+        //click on the import test case button
+        cy.get(TestCasesPage.importTestCasesBtn).click()
+
+        //select file
+        cy.get(TestCasesPage.filAttachDropBox).attachFile(singleTestCaseFile)
+
+        //import moadal should contain test case name
+        cy.get(TestCasesPage.importTestCaseModalList).should('contain.text', 'EncounterStatusMultiPerson')
+
+        //click on the 'Import' button on the modal window
+        TestCasesPage.clickImportTestCaseButton()
+
+        //test case list table contains the group name of the test case that was imported
+        cy.get(TestCasesPage.testCaseListTable).should('contain.text', 'EncounterStatusMulti')
+        //test case list table contains the title of the test case that was imported
+        cy.get(TestCasesPage.testCaseListTable).should('contain.text', 'Person')
+
+        //click on action button
+        cy.get(TestCasesPage.actionBtnNoId).eq(0).click()
+        //click on edit 
+        cy.get(TestCasesPage.editBtnNoId).contains('edit').click()
+
+        //get current web address and use that to form API request to validate JSON
+        cy.window().then((win) => {
```

Comment:
I don't think you should have to do this. We have a baseurl

Reply:
As far as my approach to get the url to hit the API request, I had to do that to get the measure ID and the test case id of the measure and test case that clicking on "cy.get(TestCasesPage.actionBtnNoId).eq(i).click()" had put me into. The approach that I was taking was less about getting the base url but more about obtaining the specific measure id and test case id.